mock/cpp_mock_scenarios/src/random_scenario/random001.cpp

```diff
@@ -0,0 +1,240 @@
+      if (
+        lanelet_pose &&
+        static_cast<traffic_simulator::LaneletPose>(lanelet_pose.value()).lanelet_id == 34684 &&
+        std::abs(static_cast<traffic_simulator::LaneletPose>(lanelet_pose.value()).s) >=
+          lane_change_position) {
+        api_.requestLaneChange(entity_name, traffic_simulator::lane_change::Direction::RIGHT);
+        lane_change_requested = true;
+      }
+    };
+
+    if (api_.isInLanelet("ego", 34684, 0.1)) {
+      spawn_and_change_lane("lane_following_0", 0.0);
+    }
+
+    /// Spawn and cross pedestrian if it does not exist and ego entity does not exists on lane "34576"
+    const auto spawn_and_cross_pedestrian = [&](const auto & entity_index) {
+      std::string entity_name = "pedestrian" + std::to_string(entity_index);
+      constexpr lanelet::Id lanelet_id = 34392;
+      if (
+        !api_.entityExists(entity_name) &&
+        !api_.reachPosition(
+          "ego", api_.canonicalize(traffic_simulator::helper::constructLaneletPose(34576, 25.0)),
+          5.0)) {
+        std::normal_distribution<> offset_distribution(
+          0.0, params_.random_parameters.crossing_pedestrian.offset_variance);
+        std::uniform_real_distribution<> speed_distribution(
+          params_.random_parameters.crossing_pedestrian.min_speed,
+          params_.random_parameters.crossing_pedestrian.max_speed);
+        api_.spawn(
+          entity_name,
+          api_.canonicalize(traffic_simulator::helper::constructLaneletPose(
+            lanelet_id, 0.0, offset_distribution(engine_))),
+          getPedestrianParameters());
+        const auto speed = speed_distribution(engine_);
+        api_.requestSpeedChange(entity_name, speed, true);
+        api_.setLinearVelocity(entity_name, speed);
+      }
+      if (api_.entityExists(entity_name) && api_.getStandStillDuration(entity_name) >= 0.5) {
+        api_.despawn(entity_name);
+      }
+    };
+    for (int i = 0; i < params_.random_parameters.crossing_pedestrian.number_of_pedestrian; i++) {
+      spawn_and_cross_pedestrian(i);
+    }
+
+    [&]() {
+      const auto trigger_position =
+        api_.canonicalize(traffic_simulator::helper::constructLaneletPose(34621, 10, 0, 0, 0, 0));
+      const auto entity_name = "spawn_nearby_ego";
+      if (api_.reachPosition("ego", trigger_position, 20.0) && !api_.entityExists(entity_name)) {
+        api_.spawn(
+          entity_name,
+          api_.getMapPoseFromRelativePose(
+            "ego", geometry_msgs::build<geometry_msgs::msg::Pose>()
+                     .position(geometry_msgs::build<geometry_msgs::msg::Point>().x(10).y(-5).z(0))
+                     .orientation(geometry_msgs::msg::Quaternion())),
+          getVehicleParameters(),
+          traffic_simulator::entity::VehicleEntity::BuiltinBehavior::doNothing());
+      }
+      if (!api_.reachPosition("ego", trigger_position, 20.0) && api_.entityExists(entity_name)) {
+        api_.despawn(entity_name);
+      }
+    }();
```

Comment:
Why to use a lambda function here?
I think there is no advantage to use lambda function here than {} to limit scope.